app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -793,6 +826,25 @@ public boolean matches(UserKeyRecord ukr) {
         CommCareApplication._().closeUserSession();
     }
 
+    public String getCurrentUserId() {
+        if (currentUserId != null) {
+            return currentUserId;
+        }
+        try {
+            currentUserId = this.getSession().getLoggedInUser().getUniqueId();

```

Comment:
Haha, @phillipm specifically suggested caching this since it gets called every time I report an event. Should I revert back?.. Don't really feel strongly either way. It seemed a little bit dangerous to cache instead of just making sure it's fresh each time, but it is a lot of repetitive calls.